README.md
# krado-react

> React components using the Krado design system.

[![NPM](https://img.shields.io/npm/v/krado-react.svg)](https://www.npmjs.com/package/krado-react) [![JavaScript Style Guide](https://img.shields.io/badge/code_style-standard-brightgreen.svg)](https://standardjs.com)

## Install

```bash
npm install --save krado-react
```

## Usage

```jsx
import React, { Component } from 'react'

import MyComponent from 'krado-react'
import 'krado-react/dist/index.css'

class Example extends Component {
  render() {
    return <MyComponent />
  }
}
```

## License

MIT © [akrade](https://github.com/akrade)


CHANGELOG.md
# Changelog

## [Unreleased]

## [0.16.0] - 2023-07-14

### Changed

- Upgraded from Theme UI 0.15.7 to 0.16.0.

## [0.15.1] - 2023-06-05

### Fixed

- Fixed missing date for 0.15.0 in `CHANGELOG.md`.

## [0.15.0] - 2023-06-05

Current bugs include broken Push menu on navigation.

### Changed

- Upgraded from Storybook 6 to 7.
- Upgraded from React 17 to 18.
- Upgraded from Theme UI 12 to 15.
- Upgraded from Framer Motion 4 to 10.
- Updated other dependencies.

## [0.14.1] - 2022-03-03

### Changed

- Adjusted `Push` menu scrollbar color.

## [0.14.0] - 2022-03-03

### Added

- The ability to create dynamic themes. ([KRAD-331])
- New updated colors from the Sūsū Identity (Nsõ, Nsõ Hot, Nsõ Warm, Nsõ Cool, Nsõ Cream, Nsõ White). ([KRAD-331])

### Fixed

- `Header` tissue jittering on page load.

### Removed

- Ash, Cool Ash, Warm Ash, Cream White, and Egg Shell White colors. ([KRAD-331])

[krad-331]: https://akrade.atlassian.net/browse/KRAD-331

## [0.13.1] - 2022-02-24

### Changes

- `Header` tissue elements now have transparent backgrounds.

## [0.13.0] - 2022-02-24

### Changed

- Made level 200 Push menu's `font-size` two pixels smaller. ([KRAD-322])
- `PushDivider` is now full width and extends the `Divider` cell. ([KRAD-325])

### Fixed

- Incorrect `addHeaderOffset` prop type in `Push` cell.
- `Push` cell not having a higher `z-index` level.

[krad-322]: https://akrade.atlassian.net/browse/KRAD-322
[krad-325]: https://akrade.atlassian.net/browse/KRAD-325

## [0.12.1] - 2022-02-18

### Changed

- `Push` cell custom scroll bar now has a lighter color.

## [0.12.0] - 2022-02-18

### Changed

- Default body `font-weight` to 400. ([KRAD-303])
- For better usability, the `Panel` and `Push` menu icons are now persistent regardless of hover state. ([KRAD-315])
- `Push` menu cell has been reworked for a better experience. ([KRAD-259]).

### Fixed

- Push menu not letting you scroll all the way down when there is overflowed content. ([KRAD-248])
- Push menu casing page content to not meet the left edge of the screen. ([KRAD-317])
- Push menu causing page to jitter on page load. ([KRAD-243])

[krad-303]: https://akrade.atlassian.net/browse/KRAD-303
[krad-315]: https://akrade.atlassian.net/browse/KRAD-315
[krad-259]: https://akrade.atlassian.net/browse/KRAD-259
[krad-248]: https://akrade.atlassian.net/browse/KRAD-248
[krad-317]: https://akrade.atlassian.net/browse/KRAD-317
[krad-243]: https://akrade.atlassian.net/browse/KRAD-243

## [0.11.0] - 2022-02-14

### Added

- `Header` and `Panel` menu items now have have an boolean active state. ([KRAD-263])
- Ability to add letters inside `Avatar`. ([KRAD-283])
- Ability to add a description/subtitle to a `Push` header item. ([KRAD-282])

### Changed

- Change default font size of `Text` to `body.paragraph` variant. ([KRAD-246])
- `LinkHeading` now contains it's `id`.
- Theme font styles have been corrected and adjusted. ([KRAD-266])
- The push menu item animation has been reworked. ([KRAD-258])

### Fixed

- `HeaderAvatar` cell not pulling passing props. ([KRAD-283])

### Removed

- Content for title in `Header` being uppercase by default. ([KRAD-239])
- `text.title` color bond. ([KRAD-245])

[krad-263]: https://akrade.atlassian.net/browse/KRAD-263
[krad-246]: https://akrade.atlassian.net/browse/KRAD-246
[krad-266]: https://akrade.atlassian.net/browse/KRAD-266
[krad-258]: https://akrade.atlassian.net/browse/KRAD-258
[krad-283]: https://akrade.atlassian.net/browse/KRAD-283
[krad-282]: https://akrade.atlassian.net/browse/KRAD-282
[krad-239]: https://akrade.atlassian.net/browse/KRAD-239
[krad-245]: https://akrade.atlassian.net/browse/KRAD-245

## [0.10.0] - 2022-01-25

### Added

- `Footer` tissue. ([KRAD-133])
- "Color Bond" system for theme base colors. ([KRAD-189])

[krad-133]: https://akrade.atlassian.net/browse/KRAD-133
[krad-189]: https://akrade.atlassian.net/browse/KRAD-189

## [0.9.0] - 2022-01-14

### Added

- Implemented [Storybook's official Emotion 11 quasi-compatibility](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#emotion11-quasi-compatibility). ([KRAD-211])
- Ability to add descriptions to `PanelItem`. ([KRAD-201])

### Changed

- Default body font to "[Inter](https://fonts.google.com/specimen/Inter)." ([KRAD-238])
- `Header` now has an lower elevation. ([KRAD-222])

### Fixed

- Gave the title's letter spacing in `Header` more breathing room. ([KRAD-220])
- `PushHeader` bold font weight not applying. ([KRAD-222])
- Incorrect `PushLabel` when inside a level 100 `Push` menu. ([KRAD-222])
- `Panel` menu showing up behind other elements. ([KRAD-235])
- `HeaderIconButton` props not working. ([KRAD-213])
- `Push` menu not spanning the full height of the page. ([KRAD-233])
- `Push` menu's close button not sticking to the top of the screen on scroll. ([KRAD-233])
- Not being able to scroll down to see more menu content in `Push` menu. ([KRAD-233])

[krad-211]: https://akrade.atlassian.net/browse/KRAD-211
[krad-201]: https://akrade.atlassian.net/browse/KRAD-201
[krad-238]: https://akrade.atlassian.net/browse/KRAD-238
[krad-233]: https://akrade.atlassian.net/browse/KRAD-233
[krad-220]: https://akrade.atlassian.net/browse/KRAD-220
[krad-222]: https://akrade.atlassian.net/browse/KRAD-222
[krad-235]: https://akrade.atlassian.net/browse/KRAD-235
[krad-213]: https://akrade.atlassian.net/browse/KRAD-213

## [0.8.0] - 2022-01-03

### Added

- `Container` cell. ([KRAD-218])
- `Divider` cell. ([KRAD-219])
- Re-added `Heading` cell.
- `LinkHeading` cell.

### Changed

- Adjusted `Push` close animation for give extra space for the open/close open.

### Fixed

- `Text` turning bold when set as an heading tag like `<h2>`. ([KRAD-217])
- `Push` not exporting correctly. ([KRAD-214])

### Removed

- Unused React imports inside `Panel` and `Push`. ([KRAD-215])

[krad-218]: https://akrade.atlassian.net/browse/KRAD-218
[krad-219]: https://akrade.atlassian.net/browse/KRAD-219
[krad-217]: https://akrade.atlassian.net/browse/KRAD-217

src/theme.js
import kradoColors from './colors/colors';
import { buildColorTheme } from './colors/getColors';

const theme = {
  space: [0, 4, 8, 16, 32, 64, 128, 256, 512],
  sizes: {
    small: '40rem',
    medium: '52rem',
    large: '90rem',
    extraLarge: '120rem',
    container: '120rem'
  },
  breakpoints: ['40em', '52em', '90em', '120rem'],
  fontSizes: [12, 14, 16, 18, 20, 22, 24, 32, 42, 52, 62, 84],
  colors: buildColorTheme({
    primary: 'oshun',
    secondary: 'oshun',
    accent: 'oshun',
    error: 'asset',
    warning: 'oshun',
    success: 'usir',
    info: 'ptah',
    surface: 'nsoWarm',
    action: 'oshun',
    text: 'nsoWarm',
    background: 'nso',
    divider: 'nso'
  }),
  fonts: {
    body: 'Inter, Arial, sans-serif',
    heading: 'Inter, Arial, sans-serif',
    display: 'Merriweather, Georgia, serif'
  },
  fontWeights: {
    body: 400,
    heading: 400,
    bold: 700
  },
  lineHeights: {
    body: 1.5,
    heading: 1.125
  },
  letterSpacings: {
    normal: 'normal',
    display: '-0.05em',
    pretext: '0.22em'
  },
  zIndices: [0, 10, 20, 30, 40, 50],
  radii: [0, 4, 8, 12, 99999],
  shadows: {
    soft: {
      low: '0px 4px 24px rgba(0, 0, 0, 0.1)',
      highMiddle: '0px 20px 64px rgba(0, 0, 0, 0.08)',
      highNorth: '0px -20px 114px rgba(0, 0, 0, 0.16)',
      highSouth: '0px 40px 114px rgba(0, 0, 0, 0.16)',
      highWest: '-40px 0px 114px rgba(0, 0, 0, 0.16)',
      highEast: '40px 0px 114px rgba(0, 0, 0, 0.16)'
    },
    hard: {
      low: '0px 4px 4px rgba(0, 0, 0, 0.25)',
      high: '0px 9px 15px rgba(0, 0, 0, 0.13)'
    },
    onFocus: `0px 0px 10px 0.5px ${kradoColors.ptah500}`
  },
  borders: ['none', `2px solid black`],
  buttons: {
    solid: {
      color: 'primary.contrast',
      bg: 'primary.regular',
      '&:hover': {
        bg: 'action.hover'
      },
      '&:active': {
        bg: 'action.active'
      },
      '&:disabled': {
        color: 'action.disabled',
        bg: 'action.contrast'
      },
      '&:focus-visible': {
        outline: 'none',
        boxShadow: 'onFocus'
      }
    },
    text: {
      color: 'primary.regular',
      bg: 'transparent',
      '&:hover': {
        color: 'action.hover'
      },
      '&:active': {
        color: 'action.active'
      },
      '&:disabled': {
        color: 'action.disabled'
      }
    },
    ghost: {
      color: 'primary.regular',
      bg: 'transparent',
      border: 1,
      borderColor: 'primary.regular',
      '&:hover': {
        color: 'action.hover',
        borderColor: 'transparent',
        bg: 'background',
        boxShadow: 'hard.high'
      },
      '&:active': {
        color: 'background',
        bg: 'action.active'
      },
      '&:disabled': {
        color: 'action.disabled',
        borderColor: 'action.disabled',
        boxShadow: 'none'
      }
    },
    sizes: {
      large: { paddingX: '55px', paddingY: '20px' },
      small: { paddingX: '35px', paddingY: '15px', fontSize: '13px' }
    },
    icon: {
      width: '32px',
      height: '32px',
      borderRadius: 2,
      backgroundColor: 'background'
    }
  },
  images: {
    avatar: {
      width: 42,
      height: 42
    }
  },
  styles: {
    root: {
      fontFamily: 'body',
      fontWeight: 'body'
    },
    a: {
      color: 'primary.regular',
      borderBottom: '3px solid',
      borderColor: 'primary.regular',
      textDecoration: 'none'
    },
    hr: {
      color: 'divider'
    }
  },
  text: {
    base: {
      color: 'text.primary'
    },
    default: {
      variant: 'text.base'
    },
    body: {
      h1: {
        variant: 'text.base',
        fontSize: 8,
        lineHeight: 'heading'
      },
      h2: {
        variant: 'text.body.h1',
        fontSize: 7
      },
      h3: {
        variant: 'text.body.h1',
        fontSize: 6
      },
      h4: {
        variant: 'text.body.h1',
        fontSize: 5
      },
      h5: {
        variant: 'text.body.h1',
        fontWeight: 600,
        fontSize: 4
      },
      summary: {
        variant: 'text.base',
        fontSize: 6,
        lineHeight: '32px',
        letterSpacing: '-0.02em'
      },
      paragraph: {
        variant: 'text.base',
        fontSize: 4,
        lineHeight: '28px'
      },
      smallParagraph: {
        variant: 'text.base',
        fontSize: 2,
        lineHeight: '20px'
      },
      bold: {
        variant: 'text.base',
        fontWeight: 'bold',
        fontSize: 4,
        lineHeight: '28px'
      },
      callout: {
        variant: 'text.base',
        fontSize: 3,
        lineHeight: '20px'
      },
      caption: {
        variant: 'text.base',
        fontWeight: 500,
        fontSize: 0,
        lineHeight: '14.52px'
      },
      footnote: {
        variant: 'text.base',
        fontWeight: 600,
        fontSize: '13px',
        lineHeight: '16.34px'
      },
      pretext: {
        variant: 'text.base',
        fontWeight: 'bold',
        fontSize: 1,
        lineHeight: '18px',
        letterSpacing: '0.22em',
        textTransform: 'uppercase',
        color: 'text.secondary'
      }
    },
    display: {
      display: {
        variant: 'text.base',
        fontFamily: 'display',
        fontSize: '104px',
        lineHeight: '104px',
        letterSpacing: 'display'
      },
      h1: {
        variant: 'text.base',
        fontFamily: 'display',
        fontSize: 10,
        lineHeight: '78px',
        letterSpacing: 'display'
      },
      h2: {
        variant: 'text.display.h1',
        fontSize: 9,
        lineHeight: '65px'
      },
      h3: {
        variant: 'text.display.h1',
        fontSize: 8,
        lineHeight: '44px'
      },
      h4: {
        variant: 'text.display.h1',
        fontSize: 7,
        lineHeight: '40px'
      },
      h5: {
        variant: 'text.display.h1',
        fontSize: 5,
        lineHeight: '28px'
      },
      quote: {
        variant: 'text.base',
        fontFamily: 'display',
        fontStyle: 'italic',
        fontSize: 6,
        lineHeight: '31px'
      }
    }
  }
};

export default theme;
